esmvalcore/preprocessor/_multimodel.py

```diff
@@ -248,115 +128,179 @@ def _unify_time_coordinates(cubes):
         cube.coord('time').guess_bounds()
 
 
-def _get_time_slice(cubes, time):
-    """Fill time slice array with cubes' data if time in cube, else mask."""
-    time_slice = []
-    for cube in cubes:
-        cube_time = cube.coord('time').points
-        if time in cube_time:
-            idx = int(np.argwhere(cube_time == time))
-            subset = cube.data[idx]
-        else:
-            subset = np.ma.empty(list(cube.shape[1:]))
-            subset.mask = True
-        time_slice.append(subset)
-    return time_slice
+def _time_coords_are_aligned(cubes):
+    """Return `True` if time coords are aligned."""
+    first_time_array = cubes[0].coord('time').points
 
+    for cube in cubes[1:]:
+        other_time_array = cube.coord('time').points
+        if not np.array_equal(first_time_array, other_time_array):
+            return False
 
-def _assemble_data(cubes, statistic, span='overlap'):
-    """Get statistical data in iris cubes."""
-    # New time array representing the union or intersection of all cubes
-    time_spans = [cube.coord('time').points for cube in cubes]
-    if span == 'overlap':
-        new_times = reduce(np.intersect1d, time_spans)
-    elif span == 'full':
-        new_times = reduce(np.union1d, time_spans)
-    n_times = len(new_times)
+    return True
 
-    # Target array to populate with computed statistics
-    new_shape = [n_times] + list(cubes[0].shape[1:])
-    stats_data = np.ma.zeros(new_shape, dtype=np.dtype('float32'))
 
-    # Realize all cubes at once instead of separately for each time slice
-    _ = [cube.data for cube in cubes]
+def _subset(cube, time_points):
+    """Subset cube to given time range."""
+    begin = cube.coord('time').units.num2date(time_points[0])
+    end = cube.coord('time').units.num2date(time_points[-1])
+    constraint = iris.Constraint(time=lambda cell: begin <= cell.point <= end)
+    return cube.extract(constraint)
```

Comment:
I think a try/except is needed here with except raising a nicely explained ValueError just in case

Reply:
Done 👍